env-check.js

```diff
@@ -0,0 +1,25 @@
+// @ts-check
+/**
+ * This file is included in `/remix.config.js` which ensures the app isn't built with invalid env vars.
+ * It has to be a `.js`-file to be imported there.
+ */
+
+// Remix does not read .env files when building for production
+if (process.env.NODE_ENV !== "production") {
+  const REQUIRED_ENVS = ["DATABASE_URL", "AUTH_SECRET"];
```

Comment:
database url is not required in production?

Reply:
I think this logic is fundamentally flawed, you want to check first variables that are required in all environment, then to check the once required in specific environments, dev or prod

Reply:
Yeah it is, and that's honestly why i am having so many issues with remix, they make these decisions that make no sense and just make everyone go around in circles to fix issues for their decisions 

It's their decision to not read environment files in remix build, so the issue i can probably support this in prod by still checking BUT then it will break you running yarn build on your machine because remix will not check your env file.

It seems like all options lead to at least one problem

Reply:
So what I figured could work would be to add `dotenv` at the start of that file since remix already uses it and that works so now it works locally and in deployment

Should not be this hard